### ISSUE TITLE
`solid/reactivity` should treat functions in `mergeProps` as being inside a tracking context

### ISSUE THREAD

**Issue by fongandrew** [solidjs-community/eslint-plugin-solid#179]
**Describe the bug**
If you put a function inside `mergeProps` that calls a signal or prop, `solid/reactivity` will warn:

```
This function should be passed to a tracked scope (like createEffect) or an event handler because it contains reactivity, or else changes will be ignored.
```

However, this is not true because `mergeProps` implicitly wraps all functions with a [`createMemo`](https://github.com/solidjs/solid/blob/29ab5bb1073b0f819b4bcd2022a66e1b9537f134/packages/solid/src/render/component.ts#L205). You can verify that reactivity still happens here: https://playground.solidjs.com/anonymous/7998ef80-905a-485a-b7d2-9f7f4f4212a6

**To Reproduce**
Put a function calling a signal inside `mergeProps`:

```js
const props = mergeProps(props, () => ({ otherProp: signalAccessor() }));
```

**Expected behavior**
There is no warning.

**Screenshots**
https://github.com/user-attachments/assets/a5cad8e7-3e0e-428f-ad1b-2c60435b66a0

**Environment (please complete the following information):**

- OS: MacOS 15.3.1
- Node version (`node --version`): 23.5.9
- `eslint-plugin-solid` version (`npm list eslint-plugin-solid`/`yarn why eslint-plugin-solid`): 0.14.4
- `eslint` version (`npm list eslint`/`yarn why eslint`): 9.16.0

**Additional context**
- [x] I would be willing to contribute a PR to fix this issue (happy to dig in but this issue is also my way of getting a second opinion that functions in `mergeProps` are, in fact, safe to treat as tracking).

**Comment by fongandrew** [comment]
FWIW, adding `mergeProps` to `customReactiveFunctions` seems to work as a workaround.

**Comment by ryansolid** [comment]
Confirmed on current main: functions inside `mergeProps` still warn even though the runtime wraps them in a memo, so they're genuinely tracked. `mergeProps` (and 2.0's `merge`) should be recognized as tracked scopes for function arguments. Queued for 0.16.1.

**Comment by ryansolid** [comment]
Fixed in v0.16.1. Function arguments to `mergeProps` (and `merge` in Solid 2.0) are now treated as tracked scopes — both wrap function sources in `createMemo`, so reactive reads inside them are genuinely tracked.
